client/my-sites/plans-v2/use-detect-window-boundary.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+import { throttle } from 'lodash';
+
+/**
+ * Returns whether `elementRef` has touched/crossed the upper Window's boundary.
+ *
+ * @param {MutableRefObject} elementRef Reference to an HTMLElement
+ * @returns {boolean} Whether the element crossed the upper boundary
+ */
+const useDetectWindowBoundary = ( elementRef: MutableRefObject< HTMLDivElement | null > ) => {
+	// Indicates whether the elementRef has crossed the upper window boundary
+	const [ borderCrossed, setBorderCrossed ] = useState( false );
+
+	// Persist its initial value since it could change if the element position is changed.
+	const elementInitialOffsetTop = useRef< number | null >( null );
```

Comment:
Out of curiosity, is there a specific reason why you used `useRef` rather than `useState` here?

Reply:
In fact, we don't need neither `useRef` nor `useState` for that. It was necessary when all of this code lived inside the `PlansFilterBar` component.

To answer your question, I choose `useRef` over `useState` mainly because I didn't care about updating the value, I only cared about persisting the value across renders.